src/components/CheckBox.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+    <div class="vnm-checked-div" v-show="checked === CheckStatus.CHECKED">
+        <c-button class="vnm-check-button" disabled>
+            <vue-feather type="check" class="vnm-checked-icon" stroke-width="3"/>
+        </c-button>
+    </div>
+    <div class="vnm-checked-div" v-show="checked === CheckStatus.PARTIAL">
+        <c-button class="vnm-check-button" disabled>
+            <vue-feather type="minus" class="vnm-checked-icon" stroke-width="3"/>
+        </c-button>
+    </div>
+    <div class="vnm-unchecked-div" v-show="checked === CheckStatus.UNCHECKED">
+        <c-button class="vnm-check-button" disabled></c-button>
+    </div>
+</template>
+
+<script lang="ts">
+    import { PropType, defineComponent } from 'vue';
+    import { CheckStatus } from '../types';
+    import { CButton } from "@coreui/vue";
+    import VueFeather from "vue-feather";
+
+    export default defineComponent({
+        props: {
+            checked: {
+                type: Number as PropType<CheckStatus>
+            }
+        },
+        components: {
+            VueFeather,
+            CButton
+        },
+        setup() {
+            return {
+                CheckStatus
+            }
+        }
+    });
+</script>
+
+<style scoped>
+.vnm-checked-div, .vnm-unchecked-div {
+    margin-right: 0.5rem
+}
+
+.vnm-check-button {
+    height: 1rem !important;
+    width: 1rem !important;
+    padding: 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.vnm-checked-icon {
+    height: 0.75rem;
+    width: 0.75rem;
+    margin-top: 0rem;
+    color: aliceblue;
+}
+
+.vnm-checked-div > .btn.disabled, .btn:disabled {
+    background-color: red;
+    border-color: red !important;
```

Comment:
Is this doable without the `!important`? I wonder if that give us some issues if we want to use this somewhere else and override the styles?

Reply:
Similarly for the unchecked I guess. Is the issue that these are rendered as disabled buttons in the html and the browser/bootstrap has opinions about how to colour those? Why are the buttons disabled, and how are click events going to be handled once implemented? Are the buttons going to be enabled when behaviour is implemented, or do you just want to gain the styling which makes these look buttony without actually being working buttons (and will handle click events at the level of the enclosing div)? I'm wondering if that won't be very accessible, but not sure what you're planning.

Reply:
yep this was for the bootstrap, it was looking a little weird in hint, i have removed the disabled tags on the buttons! the `!important` should not affect anything else at all because this is in a `style scoped` tag, i think its probably good to stop anything from overriding the styles of this component